sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -73,7 +74,7 @@ import org.apache.spark.util.{CircularBuffer, Utils}
 private[hive] class HiveClientImpl(
     override val version: HiveVersion,
     sparkConf: SparkConf,
-    hadoopConf: Configuration,
+    hadoopConf: Map[String, String],
```

Comment:
@rdblue FYI. I am changing the type from `Configuration` to `Map[String, String]` because it is possible that Hadoop classes are not shared between Spark side and the `IsolatedClientLoader` side. When Hadoop classes are not shared, if we pass a `Configuration` object to `HiveClientImpl`, the class of this `Configuration` object (this class is loaded by Spark's classloader) is different from the `Configuration` class loaded by `IsolatedClientLoader`.